### ISSUE TITLE
rails tests fails with "stack level too deep" with tiny_tds 1.0.2

### ISSUE THREAD

**Issue by thoughtafter** [rails-sqlserver/tiny_tds#275]
Works fine with tiny_tds 0.7.0 but with version 1.0.2 all tests error with "stack level too deep" regardless of whether they are run from rake or from the command line like so:

````
ruby -I test/ test/models/something_test.rb
````

Tests are run on a MS SQL Server 2008 system and using the activerecord-sqlserver-adapter version 4.1.2. These being old it would not surprise me if they were part of the problem.

**Comment by metaskills** [comment]
Hmm.... not seeing that here locally or on CI via Appveyor. Few questions:

* What platform are you on?
* What version of FreeTDS did you compile against?
* Can you provide a stack trace that shows where the stack is too deep?

**Comment by thoughtafter** [comment]
This is on Ubuntu 15.10, FreeTDS 0.91-6.1. I didn't find the stack trace helpful but maybe it will mean something to you:

````
rake --trace test:models
** Invoke test:models (first_time)
** Invoke test:prepare (first_time)
** Execute test:prepare
** Execute test:models
rake aborted!
SystemStackError: stack level too deep
/var/lib/gems/2.1.0/gems/rake-11.1.2/lib/rake/task.rb:191
Tasks: TOP => test:models
````

**Comment by thoughtafter** [comment]
This might help more:
````
ruby --debug -I test/ test/models/something_test.rb  2>&1 | grep tiny
Exception `SystemStackError' at /var/lib/gems/2.1.0/gems/tiny_tds-1.0.2/lib/tiny_tds/client.rb:43 - stack level too deep
````

**Comment by metaskills** [comment]
Ah yea... TinyTDS v1 requires FreeTDS v0.95 or higher. Can you uninstall it, upgrade FreeTDS and try again?

**Comment by ginolon** [comment]
Same problem on Ubuntu, install freetds and everything works beautiful!

BUT A QUESTION: How to install freetds on Heroku?

**Comment by metaskills** [comment]
Please see #226 for Heroku. Closing this one out.

**Comment by thoughtafter** [comment]
If it's not too much effort it might be worth detecting this and raising an error when the FreeTDS version is not supported. Thank you for the information.

**Comment by metaskills** [comment]
Thanks @thoughtafter, I did just that when we released v1.0.3. See the extconf.rb in this commit for the relevant changes. https://github.com/rails-sqlserver/tiny_tds/commit/d7d3e3e0a03f034bdcc0c646497d7e2c60544e37
